ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Trace.hs

```diff
@@ -321,6 +332,47 @@ traceBlockFetchClientTerminationEventTestBlockWith pid tracer = \case
     trace = traceUnitWith tracer ("BlockFetchClient " ++ condense pid)
 
 -- * Other utilities
+terseGDDEvent :: TraceGDDEvent PeerId TestBlock -> String
+terseGDDEvent = \case
+  TraceGDDEvent {sgen = GenesisWindow sgen, bounds, candidateSuffixes, losingPeers, loeHead} ->
+    unlines $ [
```

Comment:
`unlines` will leave a trailing newline after the trace lines and maybe we prefer not having this? Also, the style used to for was that a trace event of type `TraceBlaBlaEvent` would have a function `traceBlaBlaEventTestBlockWith :: Tracer m String -> TraceBlaBlaEvent -> m ()`. With this style. we could go for:

```suggestion
traceGDDEventTestBlockWith :: Tracer m String -> TraceGDDEvent PeerId TestBlock -> m ()
traceGDDEventTestBlockWith tracer = \case
  TraceGDDEvent {sgen = GenesisWindow sgen, bounds, candidateSuffixes, losingPeers, loeHead} ->
    traceLinesWith $ [
```

or even `traceUnitLinesWith tracer "GDG"` which would avoid having to handle indentation in the list.

Reply:
:+1: That works for me.